lib/cryptoexchange/exchanges/ccex/services/market.rb

```diff
@@ -0,0 +1,50 @@
+require 'bigdecimal'
+
+module Cryptoexchange::Exchanges
+  module Ccex
+    module Services
+      class Market < Cryptoexchange::Services::Market
+        class << self
+          def supports_individual_ticker_query?
+            true
+          end
+        end
+
+        def fetch(market_pair)
+          ticker_output  = super(ticker_url(market_pair))
+          volumes_output = super(volumes_url(market_pair))
+          adapt(ticker_output, volumes_output, market_pair)
+        end
+
+        def ticker_url(market_pair)
+          name = "#{market_pair.base}-#{market_pair.target}".downcase
+          "#{Cryptoexchange::Exchanges::Ccex::Market::API_URL}/#{name}.json"
+        end
+
+        def volumes_url(market_pair)
+          target = market_pair.target.downcase
+          "#{Cryptoexchange::Exchanges::Ccex::Market::API_URL}/volume_#{target}.json"
+        end
+
+        def adapt(ticker_output, volumes_output, market_pair)
+          market = ticker_output['ticker']
+          volume = HashHelper.dig(volumes_output, 'ticker', market_pair.base.downcase, 'vol')
+
+          ticker = Ccex::Models::Ticker.new
+          ticker.base      = market_pair.base
+          ticker.target    = market_pair.target
+          ticker.market    = Ccex::Market::NAME
+          ticker.last      = market['lastprice'] ? BigDecimal.new(market['lastprice'].to_s) : nil
```

Comment:
You mean the conversion from string/float/int/nil to big decimal? Yup, I think it's a good idea.